### PR TITLE
feat(function): add deploy workflow

### PR DESCRIPTION
**Pull request by Codelax**
Closes: #2202

### CODE REVIEW

File: cmd/scw/testdata/test-all-usage-function-deploy-usage.golden [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+🟥🟥🟥 STDERR️️ 🟥🟥🟥️
+Create or fetch, upload and deploy your function
+
+USAGE:
+  scw function deploy [arg=value ...]
+
+ARGS:
+  [namespace-id]    Function Namespace ID to deploy to
+  name              Name of the function to deploy, will be used in namespace's name if no ID is provided
+  runtime            (unknown_runtime | golang | python | python3 | node8 | node10 | node14 | node16 | node17 | python37 | python38 | python39 | python310 | go113 | go117 | go118 | node18 | rust165 | go119 | python311 | php82 | node19 | go120)
```
Comment (quantumsheep) [inline cmd/scw/testdata/test-all-usage-function-deploy-usage.golden:11]: Remove `unknown_runtime`

File: cmd/scw/testdata/test-all-usage-function-deploy-usage.golden [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  zip-file          Path of the zip file that contains your code
```
Comment (quantumsheep) [inline cmd/scw/testdata/test-all-usage-function-deploy-usage.golden:12]: Is it possible to change it to `path` and be able to give a non-zip directory or file and the CLI zips it?

**Review by quantumsheep — CHANGES_REQUESTED** [review]

File: cmd/scw/testdata/test-all-usage-function-deploy-usage.golden (hunk 2 above)
Comment (Codelax) [reply cmd/scw/testdata/test-all-usage-function-deploy-usage.golden:12]: Zipping a whole folder could leads to sending files with credentials or ".git" folders.
This should be achieved with a config file and multiple globs to specify files to include

File: cmd/scw/testdata/test-all-usage-function-deploy-usage.golden (hunk 1 above)
Comment (Codelax) [reply cmd/scw/testdata/test-all-usage-function-deploy-usage.golden:11]: This is the runtime values of `function function create`, I don't plan on listing them manually, this list should be updated when expanded by generation

**Review by quantumsheep — APPROVED** [review]

**Pull request merged**